File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ui/adapters/FragmentPageAdapter.java
package com.example.miditeslacoilapp.ui.adapters;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class FragmentPageAdapter extends FragmentStatePagerAdapter {
    private Context context;
    private List<Fragment> fragments;
    private List<String> fragmentTitles;

    public FragmentPageAdapter(FragmentManager fragmentManager, Context context){
        super(fragmentManager);
        this.context = context;
        fragments = new ArrayList<>();
        fragmentTitles = new ArrayList<>();
    }

    public void addFragment(Fragment fragment, String title){
        fragments.add(fragment);
        fragmentTitles.add(title);
    }

    public Context getContext() {
        return context;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    public List<Fragment> getFragments() {
        return fragments;
    }

    public void setFragments(List<Fragment> fragments) {
        this.fragments = fragments;
    }

    public List<String> getFragmentTitles() {
        return fragmentTitles;
    }

    public void setFragmentTitles(List<String> fragmentTitles) {
        this.fragmentTitles = fragmentTitles;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }
}

File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ui/adapters/FragmentPageAdapter.kt
package com.example.miditeslacoilapp.ui.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

class FragmentPageAdapter(fragmentManager: FragmentManager?) : FragmentStatePagerAdapter(fragmentManager!!) {
    private var fragments = mutableListOf<Fragment>()
    private var fragmentTitles = mutableListOf<String>()

    fun addFragment(fragment: Fragment, title: String) {
        fragments.add(fragment)
        fragmentTitles.add(title)
    }

    fun getFragments(): List<Fragment> {
        return fragments
    }

    fun setFragments(fragments: MutableList<Fragment>) {
        this.fragments = fragments
    }

    fun getFragmentTitles(): List<String> {
        return fragmentTitles
    }

    fun setFragmentTitles(fragmentTitles: MutableList<String>) {
        this.fragmentTitles = fragmentTitles
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return fragments.size
    }
}